hta/configs/event_args_yaml_parser.py

```diff
@@ -35,70 +35,69 @@ def from_string(version_str: str) -> "YamlVersion":
+] = lambda available_args: [
+    available_args[k] for k in ["info::labels", "info::name", "info::sort_index"]
+]
+ARGS_COMMUNICATION_FUNC: Callable[
+    [Dict[str, AttributeSpec]], List[AttributeSpec]
+] = lambda available_args: [
+    available_args[k]
+    for k in [
+        "nccl::collective_name",
+        "nccl::in_msg_nelems",
+        "nccl::out_msg_nelems",
+        "nccl::dtype",
+        "nccl::group_size",
+        "nccl::rank",
+        "nccl::in_split_size",
+        "nccl::out_split_size",
     ]
-)
-ARGS_BANDWIDTH_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: [
-        available_args[k] for k in ["data::bytes", "data::bandwidth"]
-    ]
-)
-ARGS_SYNC_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: [
-        available_args[k] for k in ["cuda_sync::stream", "cuda_sync::event"]
-    ]
-)
-ARGS_MINIMUM_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: [
-        available_args[k] for k in ["cuda::stream", "correlation::cpu_gpu"]
-    ]
-)
-ARGS_COMPLETE_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: [
-        available_args[k] for k in available_args if not k.startswith("info")
-    ]
-)
-ARGS_INFO_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: [
-        available_args[k] for k in ["info::labels", "info::name", "info::sort_index"]
-    ]
-)
-ARGS_COMMUNICATION_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: [
-        available_args[k]
-        for k in [
-            "nccl::collective_name",
-            "nccl::in_msg_nelems",
-            "nccl::out_msg_nelems",
-            "nccl::dtype",
-            "nccl::group_size",
-            "nccl::rank",
-            "nccl::in_split_size",
-            "nccl::out_split_size",
-        ]
-    ]
-)
-ARGS_DEFAULT_FUNC: Callable[[Dict[str, AttributeSpec]], List[AttributeSpec]] = (
-    lambda available_args: (
-        ARGS_MINIMUM_FUNC(available_args)
-        + ARGS_BANDWIDTH_FUNC(available_args)
-        + ARGS_SYNC_FUNC(available_args)
-        + ARGS_INPUT_SHAPE_FUNC(available_args)
-        + [available_args["index::external_id"]]
-    )
+]
+ARGS_DEFAULT_FUNC: Callable[
+    [Dict[str, AttributeSpec]], List[AttributeSpec]
+] = lambda available_args: (
+    ARGS_MINIMUM_FUNC(available_args)
+    + ARGS_BANDWIDTH_FUNC(available_args)
+    + ARGS_SYNC_FUNC(available_args)
+    + ARGS_INPUT_SHAPE_FUNC(available_args)
+    + [available_args["index::external_id"]]
 )
 
 
 @lru_cache()
 def parse_event_args_yaml(version: YamlVersion) -> EventArgs:
-    local_yaml_data_filepath = str(
-        importlib.resources.files(__package__).joinpath(
-            f"event_args_{version.get_version_str()}.yaml",
-        )
-    )
+    pkg_path: Path = Path(__file__).parent
```

Comment:
Why do we have to get package path this way instead of using importlib.resources.files(__package__)?

Reply:
Both should work, but HTA has been using the same method from the start.